tests/PHPStan/Testing/TypeInferenceTestCaseTest.php

```diff
@@ -90,4 +91,14 @@ public function testFileAssertionFailedErrors(string $filePath, string $errorMes
 		$this->gatherAssertTypes($filePath);
 	}
 
+	public function testVariableOrOffsetDescription(): void
+	{
+		$filePath = __DIR__ . '/data/assert-certainty-variable-or-offset.php';
+
+		[$variableAssert, $offsetAssert] = array_values($this->gatherAssertTypes($filePath));
+
+		$this->assertSame('variable $context', $variableAssert[4]);
+		$this->assertSame("offset 'email'", $offsetAssert[4]);
```

Comment:
I did it this way, because the rest in this test did not actually ran the assertions, it only waited for thrown acceptors. See line 91 above. 

So if you think we should actually call the asserts, please tell me how, and I'll make the change. 

Then we can assert on the full error message.